tests/test_ophys/test_tools_roiextractors.py

```diff
@@ -1054,6 +1055,61 @@ def test_add_fluorescence_traces_regular_timestamps(self):
             self.assertEqual(roi_response_series[series_name].starting_time, times[0])
             self.assertEqual(roi_response_series[series_name].timestamps, None)
 
+    def test_add_fluorescence_traces_regular_timestamps_with_metadata(self):
+        """Test adding traces with regular timestamps and also metadata-specified rate."""
+        times = np.arange(0, 5)
+        segmentation_extractor = generate_dummy_segmentation_extractor(
+            num_rois=2,
+            num_frames=5,
+            num_rows=self.num_rows,
+            num_columns=self.num_columns,
+        )
+        segmentation_extractor.set_times(times)
+
+        metadata = deepcopy(self.metadata)
+        metadata["Ophys"]["Fluorescence"]["roi_response_series"][0].update(rate=1.23)
```

Comment:
I have hammered this a couple of times but I still feel that that representing stuff as lists as the metadata (instead of dicts) confuses me. Like, what is the index 0 doing there?

I added the modification for the other container which I guess from the loop below you intended. I also passed the right metadata and no the one in self that you had before. Please let me know if those were as you intended.

Reply:
> I have hammered this a couple of times but I still feel that that representing stuff as lists as the metadata (instead of dicts) confuses me. Like, what is the index 0 doing there?

Yeah I believe this is on the to-do list or we are at least aware that that zero-index is a current placeholder (here and in other places, like the actual roiextractors submodule) and will eventually be replaced with the more informative link